homeassistant/components/recorder/core.py

```diff
@@ -787,9 +791,15 @@ def _run(self) -> None:
                     # was True, we need to reinitialize the listener.
                     self.hass.add_job(self.async_initialize)
             else:
+                self.migration_in_progress = False
+                self._dismiss_migration_in_progress()
                 self._notify_migration_failed()
                 return
 
+        # Schema migration and repair is now completed
+        self.migration_in_progress = False
+        self._dismiss_migration_in_progress()
```

Comment:
Is this dismiss needed?

Reply:
I thought it was cleaner to do the dismiss unconditionally here instead of gating it

Reply:
Nevermind, added a condition